datafusion/sqllogictest/test_files/subquery.slt

```diff
@@ -189,16 +189,16 @@ Projection: t1.t1_id, __scalar_sq_1.SUM(t2.t2_int) AS t2_sum
 physical_plan
 ProjectionExec: expr=[t1_id@1 as t1_id, SUM(t2.t2_int)@0 as t2_sum]
 --CoalesceBatchesExec: target_batch_size=2
-----HashJoinExec: mode=Partitioned, join_type=Right, on=[(t2_id@1, t1_id@0)], projection=[SUM(t2.t2_int)@0, t1_id@2]
-------ProjectionExec: expr=[SUM(t2.t2_int)@1 as SUM(t2.t2_int), t2_id@0 as t2_id]
+----ProjectionExec: expr=[SUM(t2.t2_int)@1 as SUM(t2.t2_int), t1_id@2 as t1_id]
+------HashJoinExec: mode=Partitioned, join_type=Right, on=[(t2_id@0, t1_id@0)]
```

Comment:
This is good. Because, projection before hash doesn't prune table. Hence hash doesn't benefit from it. Reshuffling columns are delayed until it is necessary. However, this test has afore mentioned Projection + Coalesce + Projection stack problem also.